util/src/main/java/tech/pegasys/artemis/util/backing/tree/TreeNodeImpl.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.backing.tree;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.artemis.util.backing.tree.TreeNode.Commit;
+import tech.pegasys.artemis.util.backing.tree.TreeNode.Root;
+
+class TreeNodeImpl {
+
+  static class RootImpl implements Root {
+    private final Bytes32 root;
+
+    public RootImpl(Bytes32 root) {
+      this.root = root;
+    }
+
+    @Override
+    public Bytes32 getRoot() {
+      return root;
+    }
+
+    @Override
+    public String toString() {
+      Bytes trimmed = root.trimLeadingZeros();
```

Comment:
Why trim leading zeroes at all? Can we not just print the 32 bytes in hex?

Reply:
Yep, this is debugging artifact. Fixed.